### PR TITLE
Design/waypoint orientation

### LINKED ISSUE #598: Design Document for Waypoint Orientation Issue
AC: Write a design document that solves the issue #596

### PR DESCRIPTION
**Pull request by matthewhannay567**
# Description

Contains the design document solving the problem of waypoint orientation, as well as merging the waypoint_source and action_server files.

Fixes #598

### CODE REVIEW

File: documents/design/waypoint_orientation.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Another aspect that can be fixed is to merge the action_server.cpp into waypoint_source.cpp. Since
+the two files are solely dependent on each other, their functionality can be merged into a single
+file. This would mean that waypoint_source.cpp would have a MoveBaseClient object inside of it, and that
+it would handle the client sending both file waypoints and rviz waypoints.
+
+There will be a parameter called "reading_from_file" that defines if the node will only send waypoints 
+from a file or from rviz. If true, it will only send waypoints from a file and ignore any waypoints sent
+by rviz. If false, it will not read from any file and only send waypoints from rviz.
+
+Waypoints sent from a file will be sent with sendGoalAndWait(), and from rviz it will be sent with sendGoal().
```
Comment (oswinso) [inline documents/design/waypoint_orientation.md]: Just a nitpicking thing, but you should use backticks to make `sendGoalAndWait()` a code snippet.

Comment (matthewhannay567) [reply documents/design/waypoint_orientation.md]: Fixed.

File: documents/design/waypoint_orientation.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+an orientation of yaw = 0.
+
+To calculate the intended waypoint orientation, the program will calculate the orientation that
+points the robot towards the next waypoint. For the final waypoint, the final orientation should
+correspond to one with yaw = 0.
+
+Another aspect that can be fixed is to merge the action_server.cpp into waypoint_source.cpp. Since
+the two files are solely dependent on each other, their functionality can be merged into a single
+file. This would mean that waypoint_source.cpp would have a MoveBaseClient object inside of it, and that
+it would handle the client sending both file waypoints and rviz waypoints.
```
Comment (oswinso) [inline documents/design/waypoint_orientation.md]: I think you could make this section a bit more clear by explicitly listing out what the new merged node will do, ie.

This new action_server node will do the following:
- fafwwef
- fafewfewaf
- ewaewrwa

**Comment by oswinso** [comment]
Actually, we need a bit more thought about how this will work out, as per discussions during the meeting. Specifically, the 4th waypoint to the 1st waypoint. Though, thinking about this a bit more, I think we can do some jank solution where we abuse the fact that we know the approximate locations of the wayponts, and have 1st point to 2nd, 2nd point to 3rd, and then 3rd point off of the 2nd.

**Pull request merged**